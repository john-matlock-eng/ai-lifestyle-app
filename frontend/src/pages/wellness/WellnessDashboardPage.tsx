import React, { useState, useMemo, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import GoalList from '../../features/goals/components/display/GoalList';
import { useGoals } from '../../features/goals/hooks/useGoals';
import QuickLogModal from '../../features/goals/components/logging/QuickLogModal';
import type { Goal } from '../../features/goals/types/api.types';
import HeroGreeting from '../../features/wellness/components/HeroGreeting';
import StatTile from '../../features/wellness/components/StatTile';
import GoalCardMini from '../../features/wellness/components/GoalCardMini';
import EmptyGoalsIllustration from '../../features/wellness/components/EmptyGoalsIllustration';
import AIInsightBanner from '../../features/wellness/components/AIInsightBanner';
import { CheckCircle, Flame, PercentCircle, TrendingUp } from 'lucide-react';
import confetti from 'canvas-confetti';

const WellnessDashboardPage: React.FC = () => {
  const navigate = useNavigate();
  const { data, isLoading } = useGoals({ status: ['active'] });
  const [quickLogGoalId, setQuickLogGoalId] = useState<string | null>(null);

  const goals = data?.goals || [];

  const stats = useMemo(() => {
    const active = goals.length;
    const longestStreak = goals.reduce(
      (m, g) => Math.max(m, g.progress.currentStreak || 0),
      0
    );
    const avgSuccess =
      active > 0
        ? Math.round(
            goals.reduce((sum, g) => sum + g.progress.successRate, 0) / active
          )
        : 0;
    const avgCompletion =
      active > 0
        ? Math.round(
            goals.reduce((sum, g) => sum + g.progress.percentComplete, 0) /
              active
          )
        : 0;
    return { active, longestStreak, avgSuccess, avgCompletion };
  }, [goals]);

  const handleQuickLog = (goalId: string) => {
    const goal: Goal | undefined = goals.find((g) => g.goalId === goalId);
    const maybeType = goal as unknown as { goalType?: string } | undefined;
    if (maybeType?.goalType === 'journal') {
      navigate('/journal');
      return;
    }
    setQuickLogGoalId(goalId);
  };

  useEffect(() => {
    if (
      goals.length > 0 &&
      goals.every(
        (g) => g.schedule?.frequency === 'daily' && g.progress.percentComplete >= 100
      )
    ) {
      confetti();
    }
  }, [goals]);

  return (
    <div className="min-h-screen bg-gray-50 py-8">
      <div className="max-w-5xl mx-auto space-y-8">
        <HeroGreeting />
        <AIInsightBanner />
        <div className="grid grid-cols-2 sm:grid-cols-4 gap-4">
          <StatTile icon={CheckCircle} label="Active Goals" value={stats.active} />
          <StatTile icon={Flame} label="Longest Streak" value={stats.longestStreak} />
          <StatTile icon={PercentCircle} label="Avg Success" value={`${stats.avgSuccess}%`} />
          <StatTile
            icon={TrendingUp}
            label="Today"
            value={`${stats.avgCompletion}%`}
            progress={stats.avgCompletion}
          />
        </div>
        {isLoading ? (
          <GoalList goals={[]} isLoading />
        ) : goals.length === 0 ? (
          <EmptyGoalsIllustration />
        ) : (
          <div className="space-y-3">
            {goals.map((g) => (
              <GoalCardMini key={g.goalId} goal={g} onQuickLog={handleQuickLog} />
            ))}
          </div>
        )}
      </div>
      {quickLogGoalId && (
        <QuickLogModal
          goalId={quickLogGoalId}
          isOpen={!!quickLogGoalId}
          onClose={() => setQuickLogGoalId(null)}
        />
      )}
    </div>
  );
};

export default WellnessDashboardPage;
