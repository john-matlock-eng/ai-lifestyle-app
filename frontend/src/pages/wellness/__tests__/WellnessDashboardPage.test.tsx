import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import WellnessDashboardPage from '../WellnessDashboardPage';

const queryClient = new QueryClient({
  defaultOptions: { queries: { retry: false } },
});

const renderPage = () =>
  render(
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <WellnessDashboardPage />
      </BrowserRouter>
    </QueryClientProvider>
  );

describe('WellnessDashboardPage', () => {
  it('renders hero and stat tiles', () => {
    const { asFragment } = renderPage();
    expect(screen.getByText(/welcome back/i)).toBeInTheDocument();
    expect(screen.getAllByTestId('stat-tile').length).toBeGreaterThanOrEqual(4);
    expect(asFragment()).toMatchSnapshot();
  });
});
