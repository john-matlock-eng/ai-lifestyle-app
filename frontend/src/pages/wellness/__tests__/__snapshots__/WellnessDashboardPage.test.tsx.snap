// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WellnessDashboardPage > renders hero and stat tiles 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-gray-50 py-8"
  >
    <div
      class="max-w-5xl mx-auto space-y-8"
    >
      <div
        class="text-center py-6 space-y-1"
      >
        <p
          class="text-sm text-gray-500"
        >
          Friday, July 11
        </p>
        <h1
          class="text-3xl font-bold text-[var(--color-text)]"
        >
          Welcome back
        </h1>
        <p
          class="text-gray-600 italic"
        >
          “Consistency builds habits, habits build wellness.”
        </p>
      </div>
      <div
        class="relative p-4 rounded-lg bg-[var(--color-surface)] text-[var(--color-text)] shadow"
      >
        <button
          aria-label="Dismiss insights"
          class="absolute top-2 right-2 text-gray-500 hover:text-gray-700"
        >
          <svg
            class="lucide lucide-x w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
        <h3
          class="font-semibold mb-1"
        >
          AI Insight
        </h3>
        <p
          class="text-sm"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="grid grid-cols-2 sm:grid-cols-4 gap-4"
      >
        <div
          class="flex flex-col items-center p-4 rounded-xl shadow bg-[var(--color-surface)]"
          data-testid="stat-tile"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-circle-check-big w-8 h-8 text-primary-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.801 10A10 10 0 1 1 17 3.335"
            />
            <path
              d="m9 11 3 3L22 4"
            />
          </svg>
          <div
            class="mt-2 text-xl font-bold text-[var(--color-text)]"
          >
            0
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Active Goals
          </div>
        </div>
        <div
          class="flex flex-col items-center p-4 rounded-xl shadow bg-[var(--color-surface)]"
          data-testid="stat-tile"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-flame w-8 h-8 text-primary-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
            />
          </svg>
          <div
            class="mt-2 text-xl font-bold text-[var(--color-text)]"
          >
            0
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Longest Streak
          </div>
        </div>
        <div
          class="flex flex-col items-center p-4 rounded-xl shadow bg-[var(--color-surface)]"
          data-testid="stat-tile"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-circle-percent w-8 h-8 text-primary-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="m15 9-6 6"
            />
            <path
              d="M9 9h.01"
            />
            <path
              d="M15 15h.01"
            />
          </svg>
          <div
            class="mt-2 text-xl font-bold text-[var(--color-text)]"
          >
            0%
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Avg Success
          </div>
        </div>
        <div
          class="flex flex-col items-center p-4 rounded-xl shadow bg-[var(--color-surface)]"
          data-testid="stat-tile"
        >
          <div
            class="relative inline-flex items-center justify-center "
          >
            <svg
              aria-label="Progress: 0%"
              class="transform -rotate-90"
              height="48"
              width="48"
            >
              <circle
                cx="24"
                cy="24"
                fill="none"
                r="22"
                stroke="#E5E7EB"
                stroke-width="4"
              />
              <circle
                class="transition-all duration-500 ease-out"
                cx="24"
                cy="24"
                fill="none"
                r="22"
                stroke="var(--color-accent-start)"
                stroke-dasharray="138.23007675795088"
                stroke-dashoffset="138.23007675795088"
                stroke-linecap="round"
                stroke-width="4"
              />
            </svg>
            <div
              class="absolute flex flex-col items-center justify-center"
            >
              <span
                class="text-xs font-medium"
                style="color: var(--color-accent-start);"
              >
                0%
              </span>
            </div>
          </div>
          <div
            class="mt-2 text-xl font-bold text-[var(--color-text)]"
          >
            0%
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Today
          </div>
        </div>
      </div>
      <div
        class="space-y-4"
      >
        <div
          class="bg-white rounded-lg shadow-sm border-2 p-4 animate-pulse"
        >
          <div
            class="flex items-start justify-between"
          >
            <div
              class="flex-1 space-y-3"
            >
              <div
                class="flex items-center gap-2"
              >
                <div
                  class="w-8 h-8 bg-gray-200 rounded-full"
                />
                <div
                  class="h-5 bg-gray-200 rounded w-48"
                />
              </div>
              <div
                class="h-4 bg-gray-200 rounded w-full"
              />
              <div
                class="h-2 bg-gray-200 rounded-full w-full"
              />
              <div
                class="flex gap-4"
              >
                <div
                  class="h-3 bg-gray-200 rounded w-16"
                />
                <div
                  class="h-3 bg-gray-200 rounded w-20"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="bg-white rounded-lg shadow-sm border-2 p-4 animate-pulse"
        >
          <div
            class="flex items-start justify-between"
          >
            <div
              class="flex-1 space-y-3"
            >
              <div
                class="flex items-center gap-2"
              >
                <div
                  class="w-8 h-8 bg-gray-200 rounded-full"
                />
                <div
                  class="h-5 bg-gray-200 rounded w-48"
                />
              </div>
              <div
                class="h-4 bg-gray-200 rounded w-full"
              />
              <div
                class="h-2 bg-gray-200 rounded-full w-full"
              />
              <div
                class="flex gap-4"
              >
                <div
                  class="h-3 bg-gray-200 rounded w-16"
                />
                <div
                  class="h-3 bg-gray-200 rounded w-20"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="bg-white rounded-lg shadow-sm border-2 p-4 animate-pulse"
        >
          <div
            class="flex items-start justify-between"
          >
            <div
              class="flex-1 space-y-3"
            >
              <div
                class="flex items-center gap-2"
              >
                <div
                  class="w-8 h-8 bg-gray-200 rounded-full"
                />
                <div
                  class="h-5 bg-gray-200 rounded w-48"
                />
              </div>
              <div
                class="h-4 bg-gray-200 rounded w-full"
              />
              <div
                class="h-2 bg-gray-200 rounded-full w-full"
              />
              <div
                class="flex gap-4"
              >
                <div
                  class="h-3 bg-gray-200 rounded w-16"
                />
                <div
                  class="h-3 bg-gray-200 rounded w-20"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
